src/ralph/backends/data/base.py

```diff
@@ -144,25 +144,65 @@ def list(
         """
 
 
-class BaseDataBackend(ABC):
+def get_backend_generic_argument(backend_class: Type, position: int) -> Optional[Type]:
+    """Return the generic argument of `backend_class` at specified `position`."""
+    if not hasattr(backend_class, "__orig_bases__"):
+        return None
+
+    bases = backend_class.__orig_bases__[0]
```

Comment:
Good question) Getting `__orig_bases__[0]` seems to work with the latest Python 3.7 version; however, I'm not sure whether this might work with older versions in the 3.7 line( I think it's not documented(

Also, not all backends seem compatible with python3.7 - ex.: ES & LDP backends import `Litteral`, Swift backend uses `@cached_property`, etc.

I would vote for dropping python3.7 support for backends.
Currently, we don't test the backends with Python 3.7; thus, we cannot guarantee that they still work, right?

(Also, then we could replace `__orig_bases__[0]` with the safer `typing.get_args(tp)`)
WDYT?